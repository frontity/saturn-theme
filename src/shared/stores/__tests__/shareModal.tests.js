import base from '../index';
import shareModal from '../shareModal';

describe('Theme › Shared › Stores › ShareModal', () => {
  test('Initial snapshot has not changed', () => {
    const self = shareModal.create();

    expect(self).toMatchSnapshot();
  });

  test('open() changes value of isOpen to true', () => {
    const self = base.create();
    const type = 'post';
    const id = 1;

    Object.defineProperty(self.shareModal, 'open', {
      writable: true,
      value: jest.fn(self.shareModal.open),
    });

    Object.defineProperty(self, 'share', {
      writable: true,
      value: {
        all: {
          requestCount: jest.fn(),
        },
      },
    });

    expect(self.shareModal.isOpen).toBe(false);

    self.shareModal.open({ type, id });

    expect(self.shareModal.isOpen).toBe(true);
    expect(self).toMatchSnapshot();
  });

  test('open() changes item values', () => {
    const self = base.create();
    const type = 'post';
    const id = 1;

    Object.defineProperty(self.shareModal, 'open', {
      writable: true,
      value: jest.fn(self.shareModal.open),
    });

    Object.defineProperty(self, 'share', {
      writable: true,
      value: {
        all: {
          requestCount: jest.fn(),
        },
      },
    });

    expect(self.shareModal.item).toEqual({ type: undefined, id: undefined });

    self.shareModal.open({ type, id });

    expect(self.shareModal.item).toEqual({ type: 'post', id: 1 });
    expect(self).toMatchSnapshot();
  });

  test('open() calls share.all.requestCount() with the right values', () => {
    const self = base.create();
    const type = 'post';
    const id = 1;

    Object.defineProperty(self.shareModal, 'open', {
      writable: true,
      value: jest.fn(self.shareModal.open),
    });

    Object.defineProperty(self, 'share', {
      writable: true,
      value: {
        all: {
          requestCount: jest.fn(),
        },
      },
    });

    self.shareModal.open({ type, id });

    expect(self.share.all.requestCount).toHaveBeenCalledWith({
      type: 'post',
      id: 1,
    });
  });

  test('close() changes value of isOpen to false', () => {
    const self = shareModal.create({ isOpen: true });

    Object.defineProperty(self, 'close', {
      writable: true,
      value: jest.fn(self.close),
    });

    expect(self.isOpen).toBe(true);

    self.close();

    expect(self.isOpen).toBe(false);
  });

  test('setLinkCopied() changes the value of isLinkCopied to true, and after one second to false', async () => {
    const delay = jest.fn(() => Promise.resolve());
    const self = shareModal.create(
      {},
      {
        theme: {
          delay,
        },
      },
    );

    Object.defineProperty(self, 'setLinkCopied', {
      writable: true,
      value: jest.fn(self.setLinkCopied),
    });

    expect(self.isLinkCopied).toBe(false);

    const promise = self.setLinkCopied();

    expect(self.isLinkCopied).toBe(true);

    await promise;

    expect(delay).toHaveBeenCalledWith(1000);
    expect(self.isLinkCopied).toBe(false);
  });
});
