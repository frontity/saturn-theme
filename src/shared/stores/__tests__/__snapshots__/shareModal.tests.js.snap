// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme › Shared › Stores › ShareModal Initial snapshot has not changed 1`] = `
Object {
  "isLinkCopied": false,
  "isOpen": false,
  "item": Object {
    "id": undefined,
    "type": undefined,
  },
}
`;

exports[`Theme › Shared › Stores › ShareModal open() changes item values 1`] = `
Object {
  "commentsMap": Object {},
  "h2r": Object {},
  "lang": Object {
    "code": "en",
    "current": Object {},
    "default": Object {
      "classicVersion": "Classic Version",
      "comments": "Comments",
      "copiedLink": "Copied",
      "copyLink": "Copy link",
      "gdpr": "Privacy settings",
      "interestedPosts": "You might be interested in...",
      "loadMore": "Load more",
      "loading": "Loading...",
      "moreInCategory": "More in #category#",
      "next": "Next",
      "nextPosts": "Next posts",
      "notifications": "Notifications",
      "poweredBy": "Powered by",
      "readingTime": "#number# min",
      "readingTimePlural": "#number# min",
      "share": "Share",
      "shares": "Share",
      "sharesPlural": "Shares",
      "sharesWithNumber": "#number# share",
      "sharesWithNumberPlural": "#number# shares",
      "writtenBy": "Written by",
    },
  },
  "menu": Object {
    "isOpen": false,
  },
  "scroll": Object {
    "isBarHidden": false,
    "latestScroll": 0,
    "routeChanged": false,
  },
  "share": Object {
    "all": Object {
      "countsMap": Object {},
    },
    "email": Object {
      "countsMap": Object {},
    },
    "facebook": Object {
      "countsMap": Object {},
    },
    "googlePlus": Object {
      "countsMap": Object {},
    },
    "linkedin": Object {
      "countsMap": Object {},
    },
    "pinterest": Object {
      "countsMap": Object {},
    },
    "telegram": Object {
      "countsMap": Object {},
    },
    "twitter": Object {
      "countsMap": Object {},
    },
    "whatsapp": Object {
      "countsMap": Object {},
    },
  },
  "shareModal": Object {
    "isLinkCopied": false,
    "isOpen": true,
    "item": Object {
      "id": 1,
      "type": "post",
    },
  },
}
`;

exports[`Theme › Shared › Stores › ShareModal open() changes value of isOpen to true 1`] = `
Object {
  "commentsMap": Object {},
  "h2r": Object {},
  "lang": Object {
    "code": "en",
    "current": Object {},
    "default": Object {
      "classicVersion": "Classic Version",
      "comments": "Comments",
      "copiedLink": "Copied",
      "copyLink": "Copy link",
      "gdpr": "Privacy settings",
      "interestedPosts": "You might be interested in...",
      "loadMore": "Load more",
      "loading": "Loading...",
      "moreInCategory": "More in #category#",
      "next": "Next",
      "nextPosts": "Next posts",
      "notifications": "Notifications",
      "poweredBy": "Powered by",
      "readingTime": "#number# min",
      "readingTimePlural": "#number# min",
      "share": "Share",
      "shares": "Share",
      "sharesPlural": "Shares",
      "sharesWithNumber": "#number# share",
      "sharesWithNumberPlural": "#number# shares",
      "writtenBy": "Written by",
    },
  },
  "menu": Object {
    "isOpen": false,
  },
  "scroll": Object {
    "isBarHidden": false,
    "latestScroll": 0,
    "routeChanged": false,
  },
  "share": Object {
    "all": Object {
      "countsMap": Object {},
    },
    "email": Object {
      "countsMap": Object {},
    },
    "facebook": Object {
      "countsMap": Object {},
    },
    "googlePlus": Object {
      "countsMap": Object {},
    },
    "linkedin": Object {
      "countsMap": Object {},
    },
    "pinterest": Object {
      "countsMap": Object {},
    },
    "telegram": Object {
      "countsMap": Object {},
    },
    "twitter": Object {
      "countsMap": Object {},
    },
    "whatsapp": Object {
      "countsMap": Object {},
    },
  },
  "shareModal": Object {
    "isLinkCopied": false,
    "isOpen": true,
    "item": Object {
      "id": 1,
      "type": "post",
    },
  },
}
`;
